sksurv/metrics.py

```diff
@@ -103,32 +102,34 @@ def _get_comparable(event_indicator, event_time, order):
         censored_at_same_time = ~event_at_same_time
         for j in range(i, end):
             if event_indicator[order[j]]:
-                mask = np.zeros(n_samples, dtype=bool)
-                mask[end:] = True
-                # an event is comparable to censored samples at same time point
-                mask[i:end] = censored_at_same_time
-                comparable[j] = mask
+                mask_info = (i, end, censored_at_same_time)
                 tied_time += censored_at_same_time.sum()
+                yield (j, mask_info, tied_time)
```

Comment:
Actually ... with the `yield` here then it should be possible to return a mask, not mask info, let me go try that ...
```suggestion
                yield (j, mask, tied_time)
```